src/app/operators/dialogs/csv/csv-config/csv-table/csv-table.component.spec.ts

```diff
@@ -0,0 +1,178 @@
+    });
+
+    it('should parse and update correctly', () => {
+        service.changeDataProperties({
+            delimiter: ',',
+            decimalSeparator: '.',
+            isTextQualifier: false,
+            textQualifier: '"',
+            isHeaderRow: true,
+            headerRow: 0,
+        });
+        fixture.detectChanges();
+        expect(getHeader().length).toBe(4);
+        expect(numberOfTableColumns()).toBe(4);
+        service.changeDataProperties({
+            delimiter: ',',
+            decimalSeparator: '.',
+            isTextQualifier: true,
+            textQualifier: '"',
+            isHeaderRow: true,
+            headerRow: 0,
+        });
+        fixture.detectChanges();
+        expect(getHeader().length).toBe(3);
+        expect(numberOfTableColumns()).toBe(3);
+    });
+
+    it('resizes table columns and synchronizes header and body widths when using custom headers', async () => {
+        component.data.content = '"a,b",c,dddddddddddddd\n"1,2",3,4';
+        const TABLE_WIDTH = 4;
+        service.changeDataProperties({
+            delimiter: ',',
+            decimalSeparator: '.',
+            isTextQualifier: false,
+            textQualifier: '"',
+            isHeaderRow: false,
+            headerRow: 0,
+        });
+        fixture.detectChanges();
+        let header_row = headerRow();
+        let body_row = bodyRow();
+        expect(tableWidthWithoutSpacers(header_row)).toBe(TABLE_WIDTH);
+        expect(tableWidthWithoutSpacers(body_row)).toBe(TABLE_WIDTH);
+        fixture.whenStable().then(() => {
+            for (let i = 0; i < header_row.length; i++) {
+                expect(clientWidth(header_row[i])).toBe(
+                    i % 2 === 0 ?
+                        component.cellSizes[i / 2] :
+                        component.cellSpacing
+                ); // Test whether or not the table resizes to the given values.
+                expect(clientWidth(header_row[i])).toBe(clientWidth(body_row[i]));
+                // Test if body and header have synchronized cell widths.
+            }
+        });
+    });
+
+    @Component({
+        selector: 'wave-table-test-host',
+        template: `<wave-csv-table
+            #csvTable
+            [data]="data"
+            [cellSpacing]="10"
+            [linesToParse]="15"
+        ></wave-csv-table>`,
+        providers: [CsvPropertiesService]
+    })
+    class TestHostComponent {
+        @ViewChild('csvTable') csvTable: CsvTableComponent;
+        modifiedDate = new Date();
+        data = {
+            file: new File(['"a,b",c,d\n"1,2",3,4'], 'test-file.csv', {lastModified : (new Date()).getDate(), type: 'csv'}),
+            content: '"a,b",c,d\n"1,2",3,4',
+            progress: 100,
+            isNull: false
+        };
+    }
+
+    /**=====================================================================================================================================
```

Comment:
I think you can remove the *global* comments since it clearly differs from the describe cases. It is also error prone if someone reorders the functions.